asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
Just implemented and pushed this, although it's not quite the same as proposed. Hopefully the test explains the reasons for that.